step-04/index.js

```diff
@@ -29,15 +29,15 @@ io.sockets.on('connection', function(socket) {
   socket.on('create or join', function(room) {
     log('Received request to create or join room ' + room);
 
-    var numClients = io.sockets.sockets.length;
+    var clientsInRoom = io.sockets.adapter.rooms[room];
```

Comment:
Interesting because it works fine for me. `socket.join(room);` creates a room with the given room name.  The rooms with the unique IDs are the rooms that correspond to the user right? Each user, that connects to Socket.IO, is assigned a room with a unique ID that corresponds to the user ID. Am I missing something?